src/hotspot/share/gc/shared/jvmFlagConstraintsGC.cpp

```diff
@@ -337,11 +337,11 @@ JVMFlag::Error MinTLABSizeConstraintFunc(size_t value, bool verbose) {
                         value, ThreadLocalAllocBuffer::alignment_reserve_in_bytes());
     return JVMFlag::VIOLATES_CONSTRAINT;
   }
-  if (value > (ThreadLocalAllocBuffer::max_size() * HeapWordSize)) {
+  if (value / HeapWordSize > ThreadLocalAllocBuffer::max_size()) {
     JVMFlag::printError(verbose,
-                        "MinTLABSize (" SIZE_FORMAT ") must be "
-                        "less than or equal to ergonomic TLAB maximum (" SIZE_FORMAT ")\n",
-                        value, ThreadLocalAllocBuffer::max_size() * HeapWordSize);
+                        "MinTLABSize (" SIZE_FORMAT " bytes) must be "
+                        "less than or equal to ergonomic TLAB maximum (" SIZE_FORMAT " words)\n",
```

Comment:
Wait, let's not mix "bytes" and "words" in the same message. Users do not readily know what is the HeapWordSize on the machine, or even how bytes and words might be related.

Is there a reason to even change these? Does the original code overflow?

Reply:
Yeah, IIRC I changed this to deal with overflow (on 32-bit)